errors/testing.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import "testing"
+
+// AssertKind asserts that got is of same error kind as want.
+func AssertKind(t *testing.T, got, want error) {
+	t.Helper()
+	if (got == nil) != (want == nil) {
+		t.Fatalf("got error[%v] differs from want[%v]", got, want)
+	}
+	if want == nil {
+		return
+	}
+	e1, ok := got.(*Error)
+	if !ok {
+		t.Fatal("got is not an *errors.Error")
+	}
+
+	e2, ok := want.(*Error)
+	if !ok {
+		t.Fatal("want is not an *errors.Error")
+	}
+
+	AssertIsKind(t, e1, e2.Kind)
+}
+
+// AssertIsKind asserts err is of kind k.
+func AssertIsKind(t *testing.T, err error, k Kind) {
+	t.Helper()
+	if !IsKind(err, k) {
+		t.Fatalf("error[%v] is not of kind %q", err, k)
+	}
+}
+
+// Assert err is (contains, wraps, etc) target.
+func Assert(t *testing.T, err, target error) {
```

Comment:
```suggestion
// AssertIs err is (contains, wraps, etc) target.
func AssertIs(t *testing.T, err, target error) {
```

I think just assert is too vague, specially since this is a specialized assertion built on top of "is" behavior (not some simple boolean check etc).

Reply:
the context is in the package name: `errors.Assert(err1, err2)`.. it assert that errors match.

Reply:
hmm not so sure that is clear enough, assertions are usually about something being true (predicates) or comparing things (equality). The matching logic implemented by errors.Is (Go's and ours) is different from most classical assertions. One example is that this assertion function is not symmetrical, `errors.Assert(t, err1, err2)` != `errors.Assert(t, err2, err1)`, for a function called just Assert, even though inside the errors package, that is a little odd IMHO. But not a huge deal, just sharing my perception on this.